bundles/Billing/Resources/assets/components/ExpenseSelect.js

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { withTranslation } from 'react-i18next';
+import ContentWrapper from './ContentWrapper';
+import InvoiceEntryJiraFilter from './InvoiceEntryJiraFilter';
+import ExpenseSelectTable from './ExpenseSelectTable';
+import ButtonGroup from 'react-bootstrap/ButtonGroup';
+import Button from 'react-bootstrap/Button';
+
+const ExpenseSelect = (props) => {
+    const [filterValues, setFilterValues] = useState({
+        billedFilter: '',
+        startDateFilter: '',
+        endDateFilter: '',
+        epicFilter: '',
+        versionFilter: '',
+        categoryFilter: ''
+    });
+
+    const { t } = props;
+
+    const epics = props.expenses
+        .reduce((carry, worklog) => {
+            if (worklog.issue.epicKey && !carry.hasOwnProperty(worklog.issue.epicKey)) {
+                carry[worklog.issue.epicKey] = worklog.issue.epicName;
+            }
+            return carry;
+        }, {});
+
+    const versions = props.expenses
+        .reduce((carry, worklog) => {
+            for (let versionKey in worklog.issue.versions) {
+                if (worklog.issue.versions.hasOwnProperty(versionKey) &&
+                    !carry.hasOwnProperty(versionKey)) {
+                    carry[versionKey] = worklog.issue.versions[versionKey];
+                }
+            }
+            return carry;
+        }, {});
+
+    const categories = props.expenses
+        .reduce((carry, expense) => {
+            if (expense.expenseCategory && !carry.hasOwnProperty(expense.expenseCategory.id)) {
+                carry[expense.expenseCategory.id] = expense.expenseCategory.name;
+            }
+            return carry;
+        }, {});
+
+    const handleFilterChange = (event) => {
+        const fieldName = event.target.name;
+        const fieldVal = event.target.value;
+
+        setFilterValues({
+            ...filterValues,
+            [fieldName]: fieldVal
+        });
+    };
+
+    const filterExpenses = (item) => {
+        if (filterValues.billedFilter !== '') {
+            if (filterValues.billedFilter === 'not_billed' && item.billed) {
+                return false;
+            }
+
+            if (filterValues.billedFilter === 'billed' && !item.billed) {
+                return false;
+            }
+        }
+
+        let worklogUpdatedTimestamp = (new Date(item.date)).getTime();
+
+        if (filterValues.startDateFilter !== null && filterValues.startDateFilter !== '') {
+            let startFilterTimestamp = filterValues.startDateFilter.getTime();
+
+            if (startFilterTimestamp > worklogUpdatedTimestamp) {
+                return false;
+            }
+        }
+
+        if (filterValues.endDateFilter !== null && filterValues.endDateFilter !== '') {
+            let endDate = filterValues.endDateFilter;
+            endDate.setHours(23, 59, 59);
```

Comment:
This looks fishy. What about the second between 23:59:59 and 00:00:00? `less than` (but not equal to) in the following comparison should not require this hack.

Reply:
Point taken :)